### ISSUE TITLE
Installation conflict

### ISSUE THREAD

**Issue by wuhuateng** [OpenDriveLab/DetAny3D#7]
1.Set Python to 3.8, install Unidepth, and after running bash compile. sh, running Python./scripts/demo. py will result in an error because the Python version is too low and different from Unidepth's requirement of 3.10+.
2.I noticed that there is no unidepth item in DetAny3D's requirements. txt. Did your unidepth installation succeed, or why is there no unidepth in the requirements. txt? Is unidepth installation necessary?
3.I installed it completely according to Detany3d's requirements. txt and trained it after installation, but I still encountered some issues torch.distributed.elastic.multiprocessing.errors.ChildFailedError.    Is it because the GPU model is incorrect? May I ask what model of GPU you have?

**Comment by jjxjiaxue** [comment]
Hi,

Sorry for the confusion!

We are using A100 GPUs. You don’t have to install Unidepth explicitly, but some of its dependencies might still be required — so if any package is missing, you can install it via pip (e.g., `pip install missing_package`).

If `torchrun` is giving you errors, could you try running `deploy.py`? That might help us isolate whether the issue is related to distributed training or something else.

Happy to help further.

**Comment by wuhuateng** [comment]
Thank you sooooooooooo much! Thank you very much for your warm reply. Do I need <bash compile. sh> when installing unideep,? or do I only need to install the packages that unideep depends on without compiling compile.sh?

**Comment by jjxjiaxue** [comment]
You're very welcome!

You don’t need to compile Unidepth — it’s not required for DetAny3D. You can simply install the packages that Unidepth depends on.

Alternatively, you can try running the code directly first. If there are any missing dependencies, just install them one by one via pip (e.g., `pip install package_name`).

**Comment by wuhuateng** [comment]
Thank you very much for your help. I still got an error during training, and the location of the error is cfg.scaler.scale(loss_total).backward()， It died directly during backway() without any error message. Due to GPU limitations, I cannot set the CUDA and Torch versions too low. Here is the pip list:
absl-py                   2.3.1
addict                    2.4.0
aiofiles                  23.2.1
annotated-types           0.7.0
anyio                     4.9.0
asttokens                 3.0.0
attrs                     25.3.0
blinker                   1.9.0
certifi                   2025.7.14
charset-normalizer        3.4.2
click                     8.2.1
coloredlogs               15.0.1
comm                      0.2.2
ConfigArgParse            1.7.1
contourpy                 1.3.2
cycler                    0.12.1
dash                      3.1.1
decorator                 5.2.1
defusedxml                0.7.1
einops                    0.8.0
executing                 2.2.0
fastapi                   0.116.1
fastjsonschema            2.21.1
ffmpy                     0.6.0
filelock                  3.13.1
Flask                     3.1.1
flatbuffers               25.2.10
fonttools                 4.59.0
fsspec                    2024.6.1
gradio                    4.44.1
gradio_client             1.3.0
gradio_image_prompter     0.1.0
groundingdino             0.1.0              /home/wht/GroundingDINO
grpcio                    1.73.1
h11                       0.16.0
h5py                      3.11.0
hf-xet                    1.1.5
httpcore                  1.0.9
httpx                     0.28.1
huggingface-hub           0.33.4
humanfriendly             10.0
idna                      3.10
imageio                   2.34.1
importlib_metadata        8.7.0
importlib_resources       6.5.2
ipython                   9.4.0
ipython_pygments_lexers   1.1.1
ipywidgets                8.1.7
itsdangerous              2.2.0
jedi                      0.19.2
Jinja2                    3.1.4
joblib                    1.5.1
jsonschema                4.24.0
jsonschema-specifications 2025.4.1
jupyter_core              5.8.1
jupyterlab_widgets        3.0.15
kiwisolver                1.4.8
Markdown                  3.8.2
markdown-it-py            3.0.0
MarkupSafe                2.1.5
matplotlib                3.7.5
matplotlib-inline         0.1.7
mdurl                     0.1.2
mmcv                      2.2.0
mmengine                  0.10.7
mpmath                    1.3.0
narwhals                  1.47.0
nbformat                  5.10.4
nest-asyncio              1.6.0
networkx                  3.3
numpy                     1.23.5
nvidia-cublas-cu12        12.1.3.1
nvidia-cuda-cupti-cu12    12.1.105
nvidia-cuda-nvrtc-cu12    12.1.105
nvidia-cuda-runtime-cu12  12.1.105
nvidia-cudnn-cu12         9.1.0.70
nvidia-cufft-cu12         11.0.2.54
nvidia-curand-cu12        10.3.2.106
nvidia-cusolver-cu12      11.4.5.107
nvidia-cusparse-cu12      12.1.0.106
nvidia-nccl-cu12          2.20.5
nvidia-nvjitlink-cu12     12.1.105
nvidia-nvtx-cu12          12.1.105
onnx                      1.16.1
onnxruntime               1.18.0
open3d                    0.19.0
opencv-python             4.10.0.84
orjson                    3.11.0
packaging                 25.0
pandas                    2.0.3
parso                     0.8.4
pexpect                   4.9.0
pillow                    10.4.0
pip                       25.1
platformdirs              4.3.8
plotly                    6.2.0
prompt_toolkit            3.0.51
protobuf                  4.25.5
ptyprocess                0.7.0
pure_eval                 0.2.3
pycocotools               2.0.10
pydantic                  2.11.7
pydantic_core             2.33.2
pydub                     0.25.1
Pygments                  2.19.2
pyparsing                 3.2.3
pyquaternion              0.9.9
python-box                7.3.2
python-dateutil           2.9.0.post0
python-multipart          0.0.20
pytz                      2025.2
PyYAML                    6.0.2
referencing               0.36.2
regex                     2024.11.6
requests                  2.32.4
retrying                  1.4.0
rich                      14.0.0
rpds-py                   0.26.0
ruff                      0.12.4
safetensors               0.5.3
scikit-learn              1.7.0
scipy                     1.10.0
semantic-version          2.10.0
setuptools                78.1.1
shapely                   2.1.1
shellingham               1.5.4
six                       1.17.0
sniffio                   1.3.1
stack-data                0.6.3
starlette                 0.47.1
supervision               0.26.0
sympy                     1.13.3
tabulate                  0.9.0
tensorboard               2.20.0
tensorboard-data-server   0.7.2
termcolor                 2.4.0
terminaltables            3.1.10
threadpoolctl             3.6.0
timm                      1.0.7
tokenizers                0.21.2
tomlkit                   0.12.0
torch                     2.4.1+cu121
torchaudio                2.4.1+cu121
torchvision               0.19.1+cu121
tqdm                      4.65.2
traitlets                 5.14.3
transformers              4.53.2
trimesh                   2.35.39
triton                    3.0.0        1
typer                     0.16.0
typing_extensions         4.12.2
typing-inspection         0.4.1
tzdata                    2025.2
urllib3                   2.5.0
uvicorn                   0.35.0
wcwidth                   0.2.13
websockets                12.0
Werkzeug                  3.1.3
wheel                     0.45.1
widgetsnbextension        4.0.14
xformers                  0.0.28.post1
yapf                      0.43.0
zipp                      3.23.0
Can you see the reason for the direct death during backward()?

**Comment by jjxjiaxue** [comment]
pytorch version
